### PR TITLE
First commit of an Ephemeris binding

### PR DESCRIPTION
**Pull request by clinique**
This binding proposal is a way to add daily events to openHAB.
The things handled are : holidays around many countries, saints of the day (for France, contributions welcome !) and user defined calendar files.

Signed-off-by: Gaël L'hopital <glhopital@gmail.com>

### CODE REVIEW

**Comment by martinvw** [comment]
@JueBag fyi

**Comment by martinvw** [comment]
@kaikreuzer is the abstraction level as a binding such that it can be used for other bindings or should it be another. An use case is what @JueBag was looking for for the tankerkoening binding.

File: addons/binding/org.openhab.binding.jollyday/src/main/resources/holidays/Holidays_sotd_fr.xml [hunk 1]
```diff
@@ -0,0 +1,373 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (hakan42) [inline addons/binding/org.openhab.binding.jollyday/src/main/resources/holidays/Holidays_sotd_fr.xml]: Should the "Saint of the Day" config file not better be contributed to the upstream project? Especially as you don't have to wait for a release, as you use a SNAPSHOT version in this PR anyway...

Comment (clinique) [reply addons/binding/org.openhab.binding.jollyday/src/main/resources/holidays/Holidays_sotd_fr.xml]: Yes, it's an option.

**Comment by JueBag** [comment]
Sounds promising! 
@martinvw :Thanks for pinging me.
Two questions: 
In the statement about Jolly Day I'very read that not only country specific calendars could the used but salsa calenders specific to a region. That would be needed for Germany  ( we can't make up our mind which calendar to use for the whole country 😕). I  can'tv see that used in the binding, nor do I see a possibility to set a region in PaperUI. Did I miss something?
Do I understand the code correctly, for each holiday a thing is created? If one needs to check wether a day is a holiday, a check  over al those things needs to be cycled?

**Comment by kaikreuzer** [comment]
> @kaikreuzer is the abstraction level as a binding such that it can be used for other bindings or should it be another. An use case is what @JueBag was looking for for the tankerkoening binding.

As we had discussed on the Tankerkönig PR, it indeed feels very wrong to me to provide such a feature as a binding. Ok, you might say it is similar to the Astro binding, but this is also already "on the edge" of being acceptable...

The problem is that bindings are decoupled from each other - so the primary use case where we needed the holiday information (within the Tankerkönig binding), would not be solvable here as it would require user interaction to somehow tell the one binding, which item to use for the holiday information.

So while I do not like this binding, I very much like the functionality it provides - we should really think together about how it could be made a first class citizen in the ESH framework. Possibly it could be simply an OSGi service. Based on this, we could implement rule triggers, conditions & actions and thus make the information easily accessible at places where we need it most. Wdyt?

**Comment by martinvw** [comment]
I do agree that the core focus should be around:

> Based on this, we could implement rule triggers, conditions & actions and thus make the information easily accessible at places where we need it most. Wdyt?

But would a binding like this not already allow us to solve this? And you can use rules to push the holiday switch to other bindings like the tankerkoening binding. Or would you like to push it further in, maybe we need a whiteboard session together, to get me more up to speed on the deeper architecture and vision or can you advice certain presentations covering this?

**Comment by martinvw** [comment]
Please include the build in the parent `pom.xml` in that case it will already be tested and analysed by travis. And it will become available in [jfrog builds list](https://openhab.jfrog.io/openhab/libs-pullrequest-local/org/openhab/binding/).

**Comment by clinique** [comment]
@Juebag : yes régions are handled by Jollyday and will also be added in the capabilities of this PR. 
Currently, one thing is created per Holiday type

**Comment by wborn** [comment]
Can we close this PR @clinique? I think it is continued in https://github.com/eclipse/smarthome/pull/6227.

**Pull request closed without merging**